File: two_sum/two_sum.go
// Time complexity: O(n)
// Space complexity: O(n)
func twoSum(nums []int, target int) []int {
	hashmap := make(map[int]int)

	for index, num := range nums {
		if prevIndex, exists := hashmap[target - num]; exists {
			return  []int{prevIndex, index}
		}
		hashmap[num] = index
	}

	return []int{}
}

File: two_sum/two_sum.rs
use std::collections::HashMap;


/// Time complexity: O(n)
/// Space complexity: O(n)
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::with_capacity(nums.len());

        for i in 0..nums.len() {
            if let Some(k) = map.get(&(target - nums[i])) {
                return vec![*k as i32,  i as i32]
            }
            map.insert(nums[i], i);
        }
        
        return vec![]
    }
}
